wallet-api/src/Ledger/Value/TH.hs

```diff
@@ -48,32 +45,16 @@ makeLift ''CurrencySymbol
 currencySymbol :: Q (TExp (Int -> CurrencySymbol))
 currencySymbol = [|| CurrencySymbol ||]
 
--- | A cryptocurrency value. This is a map from 'CurrencySymbol's to a
--- quantity of that currency.
---
--- Operations on currencies are usually implemented /pointwise/. That is,
--- we apply the operation to the quantities for each currency in turn. So
--- when we add two 'Value's the resulting 'Value' has, for each currency,
--- the sum of the quantities of /that particular/ currency in the argument
--- 'Value'. The effect of this is that the currencies in the 'Value' are "independent",
--- and are operated on separately.
---
--- Whenever we need to get the quantity of a currency in a 'Value' where there
--- is no explicit quantity of that currency in the 'Value', then the quantity is
--- taken to be zero.
---
--- See note [Currencies] for more details.
-newtype Value = Value { getValue :: Map.Map CurrencySymbol Int }
+-- | Cryptocurrency value
+--   See note [Currencies]
+newtype Value = Value { getValue :: [(CurrencySymbol, Int)] }
```

Comment:
why are we getting rid of a proper `Map` type here to implement our own?

Reply:
Yes, this shouldn't happen. It's not a proper `Map` type, it's a wrapper around an association list, but the interface is a mock for what will hopefully exist soon.

Reply:
ok